server/datastore/mysql/vpp.go

```diff
@@ -578,8 +580,19 @@ func (ds *Datastore) GetTitleInfoFromVPPAppsTeamsID(ctx context.Context, vppApps
 }
 
 func (ds *Datastore) GetVPPAppMetadataByAdamIDAndPlatform(ctx context.Context, adamID string, platform fleet.AppleDevicePlatform) (*fleet.VPPApp, error) {
-	stmt := `SELECT va.adam_id, va.bundle_identifier, va.icon_url, va.name, va.title_id, va.platform, va.created_at, va.updated_at
```

Comment:
This changes the semantics of this function, as instead of returning one row if the VPP app is in the list at all, it now returns the first row iff the VPP app is associated with a team...and it might be the wrong team.

If we need a VPP app in a team context, I think we have a separate function for that that includes team ID, and we should use that one.

Reply:
Makes sense, added this and am working on tests for it now.